src/management/projects.ts

```diff
@@ -0,0 +1,302 @@
+    const projectInfo = await addFirebaseToCloudProject(projectId);
+    spinner.succeed();
+
+    logger.info("");
+    if (process.platform === "win32") {
+      logger.info("=== Your Firebase project is ready! ===");
+    } else {
+      logger.info("🎉🎉🎉 Your Firebase project is ready! 🎉🎉🎉");
+    }
+    logger.info("");
+    logger.info("Project information:");
+    logger.info(`   - Project ID: ${clc.bold(projectInfo.projectId)}`);
+    logger.info(`   - Project Name: ${clc.bold(projectInfo.displayName)}`);
+    logger.info("");
+    logger.info("Firebase console is available at");
+    logger.info(`https://console.firebase.google.com/project/${clc.bold(projectId)}/overview`);
+    return projectInfo;
+  } catch (err) {
+    spinner.fail();
+    throw err;
+  }
+}
+
+/**
+ * Get the user's desired project, prompting if necessary.
+ */
+export async function getOrPromptProject(options: any): Promise<FirebaseProjectMetadata> {
+  if (options.project) {
+    return await getFirebaseProject(options.project);
+  }
+  return selectProjectInteractively();
+}
+
+async function selectProjectInteractively(
+  pageSize: number = MAXIMUM_PROMPT_LIST
+): Promise<FirebaseProjectMetadata> {
+  const { projects, nextPageToken } = await getProjectPage(pageSize);
+  if (projects.length === 0) {
+    throw new FirebaseError("There is no Firebase project associated with this account.");
+  }
+  if (nextPageToken) {
+    // Prompt user for project ID if we can't list all projects in 1 page
+    return selectProjectByPrompting();
+  }
+  return selectProjectFromList(projects);
+}
+
+async function selectProjectByPrompting(): Promise<FirebaseProjectMetadata> {
+  const projectId = await promptOnce({
+    type: "input",
+    message: "Please input the project ID you would like to use:",
+  });
+
+  return await getFirebaseProject(projectId);
+}
+
+/**
+ * Presents user with list of projects to choose from and gets project information for chosen project.
+ */
+async function selectProjectFromList(
+  projects: FirebaseProjectMetadata[] = []
+): Promise<FirebaseProjectMetadata> {
+  let choices = projects.filter((p: FirebaseProjectMetadata) => !!p).map((p) => {
+    return {
+      name: p.projectId + (p.displayName ? ` (${p.displayName})` : ""),
+      value: p.projectId,
+    };
+  });
+  choices = _.orderBy(choices, ["name"], ["asc"]);
+
+  if (choices.length >= 25) {
+    utils.logBullet(
+      `Don't want to scroll through all your projects? If you know your project ID, ` +
+        `you can initialize it directly using ${clc.bold(
+          "firebase init --project <project_id>"
+        )}.\n`
+    );
+  }
+  const projectId: string = await promptOnce({
+    type: "list",
+    name: "id",
+    message: "Select a default Firebase project for this directory:",
+    choices,
+  });
+
+  const project = projects.find((p) => p.projectId === projectId);
+
+  if (!project) {
+    throw new FirebaseError("Unexpected error. Chosen project must exist");
```

Comment:
Instead of "Chosen project must exist", let's say "Project does not exist."

Reply:
Done